src/core/main.c

```diff
@@ -1998,6 +1999,10 @@ static int do_reexecute(
                         log_error_errno(r, "Failed to switch root, trying to continue: %m");
         }
 
+        r = capability_ambient_set_apply(capability_ambient_set, /* also_inherit= */ false);
+        if (r < 0)
+                return log_error_errno(r, "Failed to apply the starting ambient set: %m");
```

Comment:
This should log an error, but say "ignoring" and attempt to continue, like other non-essential calls in this function. I fully expect we'll have trouble due to LSMs with these changes, so we should avoid hard failures in these paths